packages/lib-classifier/src/store/AnnotationsStore.js

```diff
@@ -7,17 +7,24 @@ const AnnotationsStore = types
   })
   .views(self => ({
     annotation (task) {
-      return self.annotations.get(task.taskKey) || task.createAnnotation()
+      let taskAnnotation = task.createAnnotation()
+      self.annotations.forEach(annotation => {
+        if (annotation.task === task.taskKey) {
+          taskAnnotation = annotation
+        }
+      })
+      return taskAnnotation
```

Comment:
[minor][curiosity question] This block in theory should be functionally the same as the line it's replacing - get the annotation object for the specific task, and if there's no annotation object, create one.

Out of curiosity, why use `annotations.forEach(...)` instead of the simpler `annotations.get(taskKey)`? Wait, does the old line of code, `self.annotations.get(task_key_that_does_not_exist)` return a weird MobX `EmptyObjectOrWhatever` instead of `undefined` or `null`, causing the `||` operator to never fork to `task.createAnnotation()` ?

Reply:
`map.get(identifier)` can only be used with identifiers. If you try `annotations.get(taskKey)` MobX State Tree will error because `taskKey` isn't an annotation ID.

Task keys aren't identifiers any more because each task model only exists once in the tree. However, that task may have created many annotations eg. a text sub task which annotates 50 lines of text will have 50 annotations, all with the same task key.

I spent a lot of time reading and re-reading [Identifiers and References](https://mobx-state-tree.js.org/concepts/references) trying to get my head around this. An identifier must always resolve to a single, unique node in the tree, as far as I can tell.

This block is not quite the same as the original. It loops through the annotation collection and returns the last annotation for `task.taskKey`, if there are multiple annotations for that task.

Reply:
>Task keys aren't identifiers any more because each task model only exists once in the tree. 

I could change the tree so that each task can have multiple instances, one for each time you perform a task eg. one task model per line of text. However, I think that would require dismantling the classifier and rebuilding a large chunk of it. Only one task per task key is an assumption that's built quite deeply into the classifier model at the moment.

Reply:
👌 Yoiks! This was more involved than I expected. Thanks for highlighting the issue with Identifiers and Resources for MST.